### PR TITLE
Fix issue deprecated method error

### PR DESCRIPTION
**Pull request by SonYoonSeok**

### CODE REVIEW

**Comment by SonYoonSeok** [comment]
# Summary
You can see this issue #3137.

There is a deprecated method in `SpecFinishedInterceptor` and it doesn't work.

## Before
```Kotlin
val duration = Duration.milliseconds(timeInMillis() - start)
```

## After
```Kotlin
val duration = (timeInMillis() - start).milliseconds 
```

**Review by sksamuel — APPROVED** [review]

**Comment by Kantis** [comment]
I think this was a workaround for supporting Kotlin 1.5. With this change it won't be possible to write and test gradle plugins for version 7.4 and earlier. I think it should be fine since 7.5 is stable now, but it would perhaps be good to add a compatibility matrix to the readme or something

**Comment by SonYoonSeok** [comment]
Thanks for comment @Kantis !!, I know from #3116 the Gradle version has been updated to 7.5. So I think, It will be fine, right?

File: kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/spec/interceptor/SpecFinishedInterceptor.kt [hunk 1]
```diff
@@ -22,7 +23,7 @@ internal class SpecFinishedInterceptor(private val listener: TestEngineListener)
 
       // We deliberately use the deprecated method here to maintain backwards compatibility with Kotlin 1.5.
       // See https://github.com/kotest/kotest/issues/3059.
```
Comment (Kantis) [inline kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/spec/interceptor/SpecFinishedInterceptor.kt:25]: Comment is no longer relevant and should be removed

**Review by Kantis — COMMENTED** [review]
The problem wasn't the Gradle version used by Kotest itself.  See https://github.com/kotest/kotest/issues/3059

When writing Gradle plugins you're forced to use Gradle's embedded kotlin version which for < v7.5 was Kotlin 1.5.  By using the deprecated method, we could enable those use cases still. But if this is now blocking users from migrating to Kotlin 1.7.10 I think we should go ahead with this change.

Users writing Gradle plugins could perhaps upgrade to Gradle 7.5 which moves the embedded Kotlin to 1.6 which should work with this change. What do you think @charleskorn ?

**Comment by charleskorn** [comment]
I think #3137 was fixed in 5.4.2 (the issue says they were using 5.4.1). Could you please confirm that you're seeing the issue with 5.4.2?

**Comment by SonYoonSeok** [comment]
The codes related to the error has not been updated in the master branch. I think it will still throw a error @charleskorn

**Comment by charleskorn** [comment]
I'm using Kotest 5.4.2 with Kotlin 1.7.10 with no issues, and I was previously running into this problem. There's also a test in the CI suite that runs Kotest with Kotlin 1.7.

The deprecated method [has not been removed in Kotlin 1.7.10](https://github.com/JetBrains/kotlin/blob/v1.7.10/libraries/stdlib/src/kotlin/time/Duration.kt#L268), but simply marked as deprecated, so there _shouldn't_ be any issues... but I may be wrong...

For this reason, I suspect there's another issue at play in #3137 - are you able to share a sample project that demonstrates the issue?

Of course, if we want to stop supporting Kotlin 1.5 and remove the deprecated method, go ahead, but I'm not sure this is going to fix #3137.

**Comment by sksamuel** [comment]
Kotest 5 is already 1.6 only, https://github.com/kotest/kotest/blob/master/gradle/libs.versions.toml#L12.
But I agree with you @charleskorn that changing this shouldn't fix anything as the method still exists in 1.7

**Pull request closed without merging**